opt: run CheckExpr in all tests

RaduBerinde opened the pull request:
We currently run CheckExpr in race tests. These are ran infrequently
because they are very slow (and I believe logic tests are disabled
altogether with race). Various other tests like sqlsmith probably
never run in race mode.

This change adds a `crdb_test` tag which is present only when we are
running a testing target (via the makefile) and switches these checks
to be conditioned on this tag.

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/57532)
<!-- Reviewable:end -->

RaduBerinde commented:
@irfansharif I'm not sure if I need to do anything here for bazel.

irfansharif commented:
There's nothing we need to do presently, we haven't introduced custom build tags with bazel builds/tests.

RaduBerinde commented:
CC @yuzefovich there might be some overlap between this and the metamorphic build tag

mgartner reviewed (approved):
:100: 

> Reviewed 6 of 6 files at r1.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/57532#-:-MNyqpam7CXWFluKfRIX:bmi8ppe)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @irfansharif, @jordanlewis, and @rytaft)



<!-- Sent from Reviewable.io -->

rytaft reviewed (approved):
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/> this is great!

> Reviewed 6 of 6 files at r1.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/57532#-:-MNz29na49rz3l2WPsK5:bkti68n)* status: :shipit: complete! 1 of 0 LGTMs obtained (waiting on @irfansharif, @jordanlewis, and @RaduBerinde)

___
*[pkg/sql/opt/memo/check_expr_skip.go, line 17 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/57532#-MNz1z125uI9KVs-WlYw:-MNz1z125uI9KVs-WlYx:b-y8r8x2) ([raw file](https://github.com/cockroachdb/cockroach/blob/b7690666ab35f6cb3258844696e6eb1fae70d0f2/pkg/sql/opt/memo/check_expr_skip.go#L17)):*
> ```Go
> import "github.com/cockroachdb/cockroach/pkg/sql/opt"
> 
> // CheckExpr is a no-op in non-race builds.
> ```

[nit] update comment to say non-test


<!-- Sent from Reviewable.io -->

RaduBerinde reviewed (commented):
TFTRs!

bors r+

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/57532#-:-MNzBJlJB63NZVDBc900:b-tlx7ko)* status: :shipit: complete! 0 of 0 LGTMs obtained (and 1 stale) (waiting on @irfansharif, @jordanlewis, @mgartner, and @rytaft)

___
*[pkg/sql/opt/memo/check_expr_skip.go, line 17 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/57532#-MNz1z125uI9KVs-WlYw:-MNzBJE72pH0KUPWAZb8:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/b7690666ab35f6cb3258844696e6eb1fae70d0f2/pkg/sql/opt/memo/check_expr_skip.go#L17)):*
<details><summary><i>Previously, rytaft (Rebecca Taft) wrote…</i></summary><blockquote>

[nit] update comment to say non-test
</blockquote></details>

Done.


<!-- Sent from Reviewable.io -->

RaduBerinde commented:
bors r-

RaduBerinde committed:
opt: run CheckExpr in all tests

We currently run CheckExpr in race tests. These are ran infrequently
because they are very slow (and I believe logic tests are disabled
altogether with race). Various other tests like sqlsmith probably
never run in race mode.

This change adds a `crdb_test` tag which is present only when we are
running a testing target (via the makefile) and switches these checks
to be conditioned on this tag.

Release note: None

RaduBerinde commented:
bors r+
